packaging/os/apt_repository.py

```diff
@@ -321,6 +321,21 @@ def remove_source(self, line):
         source = self._parse(line, raise_if_invalid_or_disabled=True)[2]
         self._remove_valid_source(source)
 
+    @property
+    def repos_urls(self):
+        repositories = []
+        for parsed_repos in self.files.values():
+            for parsed_repo in parsed_repos:
+                enabled = parsed_repo[1]
+                source_line = parsed_repo[3]
+
+                if not enabled:
+                    continue
+
```

Comment:
Since we are comparing against the expanded ppa in the main() function, we still need to expand ppas here.  Maybe adding back the old code and fixing the reference to "i" is what's needed:

``` python
+                if source_line.startswith('ppa:'):
+                    source, ppa_owner, ppa_name = self._expand_ppa(source_line)
+                    source_line = source
```